cpp/include/rapidsmpf/streaming/core/base_chunk.hpp

```diff
@@ -0,0 +1,76 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <cstdint>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace rapidsmpf::streaming {
+
+/**
+ * @brief Common base for streaming chunks.
+ *
+ * Holds the per-chunk sequence number (for ordering) and the CUDA stream
+ * on which the chunk's resources were created.
```

Comment:
*minor*: Not all chunks will need a stream (if we're communicating non-gpu data through the channel). I suppose we can just use the default stream in that case.